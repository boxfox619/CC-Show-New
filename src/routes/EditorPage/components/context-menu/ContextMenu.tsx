import * as React from 'react';
import { MenuContainer, MenuContent, Option, ShortCut } from './styles';

interface Menu {
    label: string
    shortcut?: string
    subMenus?: Menu[]
    disabled?: boolean
    onClick?: () => void
}

interface Props {
    visible: boolean,
    menu: Menu[]
}

const createMenu = (menus: Menu[], left?: number, top?: number) => {
    return (
        <MenuContainer left={left} top={top}>
            <MenuContent>
                {menus.map(m => (
                    <Option onClick={m.onClick} disabled={m.disabled}>
                        {m.label}
                        {m.shortcut && (<ShortCut>{m.shortcut}</ShortCut>)}
                        {m.subMenus && createMenu(m.subMenus)}
                    </Option>
                ))}
            </MenuContent>
        </MenuContainer>
    )
}

export const ContextMenu: React.FC<Props> = ({visible, menu}) => {
    return createMenu(menu, 0, 0);
}