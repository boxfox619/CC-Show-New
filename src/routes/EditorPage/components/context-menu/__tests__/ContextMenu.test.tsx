import * as React from 'react';
import * as Enzyme from 'enzyme';
import { ContextMenu } from '../ContextMenu';

describe('<ContextMenu />', () => {
    const menu = [
        { label: '맨 앞으로 가져오기', shortcut: 'SHIFT + CTRL + ]', onClick: () => alert('aa') },
        { label: '앞으로 가져오기', shortcut: 'CTRL + ]', onClick: () => alert('aa') },
        { label: '뒤로 보내기', shortcut: 'CTRL + [', onClick: () => alert('aa') },
        { label: '맨 뒤로 보내기', shortcut: 'SHIFT + CTRL + [', onClick: () => alert('aa') },
    ];
    it('should match snapshot', () => {
        const wrapper = Enzyme.shallow(<ContextMenu visible={true} menu={menu} position={{ x: 10, y: 10 }} />);
        expect(wrapper).toMatchSnapshot();
    });
});