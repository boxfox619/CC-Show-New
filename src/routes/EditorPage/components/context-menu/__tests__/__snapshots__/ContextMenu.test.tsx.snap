// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContextMenu /> should match snapshot 1`] = `ShallowWrapper {}`;
